### ISSUE TITLE
Dyanmic Tree's Losing their leaves and dying?

### ISSUE THREAD

**Issue by Runburner** [ferreusveritas/Dynamic-Trees#462]
-Description of the bug-
So what I am encountering is the fact that Dynamic Tree's are losing their leaves slowly once they are generated and then all of a sudden they lose all their leaves. Shortly after that, the tree will become a stump. It looks something like this.

https://cdn.discordapp.com/attachments/567027432587264031/770041877386625034/unknown.png

Below is the mod list I am running.

  | AppleSkin | squeek502 | AppleSkin-mc1.12-1.0.14.jar |  
  | AutoRegLib | Vazkii | AutoRegLib-1.3-32.jar |  
  | Baubles | Azanor13 | Baubles-1.12-1.5.2.jar |  
  | Better Advancements | way2muchnoise | BetterAdvancements-1.12.2-0.1.0.77.jar |  
  | Better Animal Models | cybercat5555 | betteranimals-1.12.2-5.3.0.jar |  
  | Better Animals Plus | cybercat5555 | betteranimalsplus-1.12.2-9.0.1.jar |  
  | Better Builder's Wands | Portablejim | BetterBuildersWands-1.12-0.11.1.245+69d0d70.jar |  
  | Better Foliage | octarine_noise | BetterFoliage-MC1.12-2.3.1.jar |  
  | Better Ping Display | Quintinity | BetterPingDisplay-1.12.2-1.0.jar |  
  | BetterFps | Guichaguri | BetterFps-1.4.8.jar |  
  | Biomes O' Plenty | Forstride | BiomesOPlenty-1.12.2-7.0.1.2441-universal.jar |  
  | Bookshelf | DarkhaxDev | Bookshelf-1.12.2-2.3.590.jar |  
  | Chameleon | Texelsaur | Chameleon-1.12-4.1.3.jar |  
  | Chunk-Pregenerator | Speiger | Chunk Pregenerator V1.12-2.4E.jar |  
  | CoFH Core | TeamCoFH | CoFHCore-1.12.2-4.6.6.1-universal.jar |  
  | CoFH World | TeamCoFH | CoFHWorld-1.12.2-1.4.0.1-universal.jar |  
  | ConnectedTexturesMod | tterrag1098 | CTM-MC1.12.2-1.0.2.31.jar |  
  | Cooking for Blockheads | BlayTheNinth | CookingForBlockheads_1.12.2-6.5.0.jar |  
  | CoroUtil | Corosus | coroutil-1.12.1-1.2.37.jar |  
  | CraftTweaker | Jaredlll08 | CraftTweaker2-1.12-4.1.20.586.jar |  
  | Dynamic Surroundings | OreCruncher | DynamicSurroundings-1.12.2-3.6.1.0.jar |  
  | Dynamic Surroundings: HUDs | OreCruncher | DynamicSurroundingsHuds-1.12.2-3.6.0.0.jar |  
  | Dynamic Trees | ferreusveritas | DynamicTrees-1.12.2-0.9.20.jar |  
  | Dynamic Trees - Biomes O' Plenty Compat | mangoose3039 | DynamicTreesBOP-1.12.2-1.4.4.jar |  
  | Dynamic Trees - Forestry | supermassimo0310 | DynamicTreesForestry-1.12.2-1.0.9.jar |  
  | Dynamic Trees - Pam's Harvestcraft Compat | ferreusveritas | DynamicTreesPHC-1.12.2-2.0.4.jar |  
  | Dynamic Trees - Quark | supermassimo0310 | DynamicTreesQuark-1.12.2-1.0.6.jar |  
  | Dynamic Trees - Thaumcraft Compat | mangoose3039 | DynamicTreesTC-1.12.2-1.4.2.jar |  
  | Dynamic Trees - The Twilight Forest | supermassimo0310 | dynamictreesTheTwilightForest-1.12.2-1.2.2.jar |  
  | Dynamic Trees - Tinker's Construct | supermassimo0310 | DynamicTreesTinkersConstruct-1.12.2-1.2.5.jar |  
  | Dynamic Trees - Traverse | HarleyOConnor | DynamicTreesTraverse-1.12.2-2.1.jar |  
  | Farming for Blockheads | BlayTheNinth | FarmingForBlockheads_1.12.2-3.1.28.jar |  
  | Forestry | _ForgeUser7678633 | forestry_1.12.2-5.8.2.387.jar |  
  | FTB Backups | FTB | FTBBackups-1.1.0.1.jar |  
  | FTB Library | FTB | FTBLib-5.4.7.2.jar |  
  | FTB Utilities | FTB | FTBUtilities-5.4.1.130.jar |  
  | Future MC | thedarkcolour | future-mc-1.12.2-0.2.4.1.jar |  
  | Game Stages | DarkhaxDev | GameStages-1.12.2-2.0.123.jar |  
  | Hwyla | TehNut | Hwyla-1.8.26-B41_1.12.2.jar |  
  | Immersive Engineering | BluSunrize | ImmersiveEngineering-0.12-98.jar |  
  | IvToolkit | Ivorforce | IvToolkit-1.3.3-1.12.jar |  
  | JourneyMap | techbrew | journeymap-1.12.2-5.7.1.jar |  
  | Just Enough HarvestCraft (JEHC) | mrapplexz | just-enough-harvestcraft-1.12.2-1.7.2.jar |  
  | Just Enough Items (JEI) | mezz | jei_1.12.2-4.16.1.301.jar |  
  | Just Enough Resources (JER) | way2muchnoise | JustEnoughResources-1.12.2-0.9.2.60.jar |  
  | MalisisCore | Ordinastie | malisiscore-1.12.2-6.5.1.jar |  
  | MalisisDoors | Ordinastie | malisisdoors-1.12.2-7.3.0.jar |  
  | Mantle | mDiyo | Mantle-1.12-1.3.3.55.jar |  
  | Metallurgy 4: Reforged | Davoleo_K | Metallurgy-4-Reforged-1.1.2.jar |  
  | Neat | Vazkii | Neat 1.4-17.jar.disabled |  
  | OreLib | OreCruncher | OreLib-1.12.2-3.6.0.1.jar |  
  | Pam's HarvestCraft | pamharvestcraft | Pam's HarvestCraft 1.12.2zg.jar |  
  | Patchouli | Vazkii | Patchouli-1.0-22.jar |  
  | Potion Core | Tmtravlr | PotionCore-1.9_for_1.12.2.jar |  
  | Quark | Vazkii | Quark-r1.6-179.jar |  
  | Railcraft | Covert_Jaguar | railcraft-12.0.0.jar |  
  | Random Things | Lumien231 | RandomThings-MC1.12.2-4.2.7.4.jar |  
  | RandomPatches | TheRandomLabs | randompatches-1.12.2-1.22.1.7.jar |  
  | ReAuth | TechnicianLP | reauth-3.6.0.jar |  
  | Recurrent Complex | Ivorforce | RecurrentComplex-1.4.8.2.jar |  
  | Redstone Flux | TeamCoFH | RedstoneFlux-1.12-2.1.1.1-universal.jar |  
  | Roguelike Dungeons | Greymerk | RoguelikeDungeons-1.12.2-1.8.0.jar |  
  | RPG-Hud [Forge] | KurodaAkira | [1.12][Forge] RPG-HUD-3.8.1.jar |  
  | Serene Seasons | TheAdubbz | SereneSeasons-1.12.2-1.2.18-universal.jar |  
  | Shadowfacts' Forgelin | ShadowfactsDev | Forgelin-1.8.4.jar |  
  | Storage Drawers | Texelsaur | StorageDrawers-1.12.2-5.4.2.jar |  
  | SwingThroughGrass | exidex | stg-1.12.2-1.2.3.jar |  
  | Thaumcraft | Azanor13 | Thaumcraft-1.12.2-6.1.BETA26.jar |  
  | The Twilight Forest | Benimatic | twilightforest-1.12.2-3.11.1021-universal.jar |  
  | Tinkers Construct | mDiyo | TConstruct-1.12.2-2.13.0.183.jar |  
  | Traverse: Legacy Continued | ProspectorDev | Traverse-1.12.2-1.6.0-69.jar |  
  | UniDict | WanionCane | UniDict-1.12.2-2.9.8.jar |  
  | Waila Harvestability | squeek502 | WailaHarvestability-mc1.12-1.1.12.jar |  
  | Waila Stages | DarkhaxDev | WailaStages-1.12.2-2.0.23.jar |  
  | WanionLib | WanionCane | WanionLib-1.12.2-2.5.jar |  
  | Waystones | BlayTheNinth | Waystones_1.12.2-4.1.0.jar |  
  | Weather, Storms & Tornadoes | Corosus | weather2-1.12.1-2.6.12.jar

If you need more information please let me know. I've already tested disabling certain mods but have had no luck so far in pinpointing it.

**Comment by Runburner** [comment]
Adding a bit more information here. I tested again with Just Dynamic Trees, BetterFPS, and Optifine. It's still happening to all the trees. I went ahead and tried deleting the configuration files to see if that fixed it and it looks like that fixed it. Could it be that changing this line would cause that issue???

# Factor that multiplies the rate at which trees grow. Use at own risk [range: 0.0 ~ 16.0, default: 0.5]
    S:growthMultiplier=0.7

**Comment by Harleyoc1** [comment]
Sorry to be the one to inform you, but it appears Herobrine has taken over your world.

**Comment by GameDungeon** [comment]
I have not changed the config and I believe a similar thing is happening to me. Jungle trees near my base are losing all leaves. I don't have optifine, but I do have fast leaf decay. Though the leaves should not be decaying.

**Comment by Harleyoc1** [comment]
Shouldn’t be fast leaf decay as DT has specific compatibility to cancel their leaf decay events for dynamic leaves.

**Comment by COMBOhrenovke** [comment]
Then Herobrine it is.

**Comment by Lifeisascam** [comment]
Hello, I had exactly the same issue. Me and my friend were building a house in a green biome and after some time we realized that every tree around us was dead. I tried changing mod version and tweaking some config files for some time and finally found the way to solve this problem.

# The chance of a tree on depleted soil to die. 1/256(~0.004) averages to about 1 death every 16 minecraft days [range: 0.0 ~ 1.0, default: 0.0]
    S:diseaseChance=0.0

Mine was set on 0.2 after I changed it to default (0.0) and started observing trees again, none of the trees that usually died within 1 minute of looking at them died after this change. I never touched this line but it somehow set itself to 0.2. Honestly I think this function is not working as it is intended to. Hope this helps!

**Comment by Harleyoc1** [comment]
Are you playing on a mod pack that perhaps had this set? As the comment suggests the default is definitely `0.0`... Either way it seems to me that it's functioning the way it's meant to for the value that was set, if you consider the comment which says that 0.004 equates to a singular tree dying after 16 Minecraft days.

**Comment by Lifeisascam** [comment]
I am using a lot of mods that I downloaded myself. I looked at the list of mods provided by Runburner in order to test the ones we have in common. I wanted to find which mod pack changes the value from default. After downloading them one by one values stayed the same and then I decided to check my original forge profile, the one I was using is a copy of that original. In original values are 0.0. This means that change happened only on the copy of the profile. On the copied profile I was trying to adjust seed drop rates in my world and remember setting them to 0.2 but after seeing no change, I opened config again and the values on the line that is responsible for seed drop rates did not change at all. I must have put down 0.2 in the wrong line without realizing it. Maybe Runburner did the same mistake but on the other side GameDungeon has similar issue without messing with config files. By the way with diseaseChance=0.2  every tree in 8 chunk radius around me was dead by the second day in minecraft. 

Here is the list of mods I am using.

CreativeCore/CreativeMD / CreativeCore_v1.10.45_mc1.12.2.jar
SwingThroughGrass/exidex/stg-1.12.2-1.2.2.jar
☃ Snow! Real Magic!/Snownee_/SnowRealMagic-0.3.3.jar
Advanced Hook Launchers/EnderLanky/AdHooks-1.12.2-3.3.0.0-build.0528.jar
Ancient Warfare 2/Shadowmage4513/ancientwarfare-1.12.2-2.7.0.1032.jar
Anvil Patch - lawful/Lumber_Wizard/anvilpatch-1.0.1.jar
Artifacts/ochotonida/Artifacts-1.12.2-1.2.3.jar
AtomicStryker's Infernal Mobs/atomicstrykergrumpy/InfernalMobs-1.12.2.jar
Baubles/Azanor13/Baubles-1.12-1.5.2.jar
Better Animal Models/cybercat5555/betteranimals-1.12.2-5.3.1.jar
Better Animals Plus/cybercat5555/betteranimalsplus-1.12.2-9.0.1.jar
Better Foliage/octarine_noise/BetterFoliage-MC1.12-2.3.1.jar
BetterNether/paulevs/betternether-0.1.8.6.jar
Biomes O' Plenty/Forstride/BiomesOPlenty-1.12.2-7.0.1.2441-universal.jar
Block Armor/2PiRadians/BlockArmor-1.12.2-2.4.12.jar
Bloodmoon/Lumien231/Bloodmoon-MC1.12.2-1.5.3.jar
Born in a Barn/Speiger/Born In A Barn V1.8-1.12-1.1.jar
BountifulBaubles/Cursed1nferno/Bountiful Baubles-1.12.2-0.1.6.jar
Carry On/Tschipp/carryon-1.12.2-1.12.3.15.jar
Chicken Drop Feathers/baithoven21/ChickenDropFeathers-1.0-forge-1.12.2-14.23.5.2854.jar
CodeChicken Lib 1.8.+/covers1624/CodeChickenLib-1.12.2-3.2.3.358-universal.jar
Corpse/henkelmax/corpse-1.12.2-1.0.8.jar
Craftable Horse Armour & Saddle [CHA&S]/EwyBoy/CraftableHorseArmour-1.3.0-1.12.jar
Davincis Vessels/darkevilmacdavincisvessels-1.12-6.340-full.jar
Disenchanter (The Disenchanter Mod)/impelon/disenchanter[1.12]1.7.jar
Doggy Talents/percivalalb/DoggyTalents-1.12.2-1.15.1.6.jar
Dynamic Surroundings/OreCruncher/DynamicSurroundings-1.12.2-3.6.1.0.jar
Dynamic Trees/ferreusveritas/DynamicTrees-1.12.2-0.9.22.jar
Dynamic Trees - Biomes O' Plenty Compat/cadaverous_queen/DynamicTreesBOP-1.12.2-1.4.4.jar
Enchantment Descriptions/DarkhaxDev/EnchantmentDescriptions-1.12.2-1.1.15.jar
End: Overload/Forge_User_19312749/minecraft_end_overload_v1.3.0_forge1.12.2.jar
EnhancedVisuals/CreativeMD/EnhancedVisuals_v1.3.24_mc1.12.2.jar
Foam​Fix/asiekierka/foamfix-0.10.11-1.12.2.jar
Food Expansion/Lellson8/FoodExpansion1.3.3-1.12.2.jar
ForgeEndertech/EnderLanky/ForgeEndertech-1.12.2-4.5.5.0-build.0561.jar
Future MC/thedarkcolour/future-mc-1.12.2-0.2.6.1.jar
Global XP/bl4ckscor3/globalxp-1.12.2-v1.4.10.jar
IvToolkit/Ivorforce/IvToolkit-1.3.3-1.12.jar
Just Enough Items (JEI)mezz/jei_1.12.2-4.16.1.301.jar
LLibrary/_ForgeUser11902522/llibrary-1.7.20-1.12.2.jar
MovingWorld/darkevilmac/movingworld-1.12-6.353-full.jar
Mowzie's Mobs/bobmowzie/mowziesmobs-1.5.8.jar
OreLib/OreCruncher/OreLib-1.12.2-3.6.0.1.jar
Phosphor (Forge)/jellysquid_/phosphor-forge-mc1.12.2-0.2.7-universal.jar
Potion Core/Tmtravlr/PotionCore-1.9_for_1.12.2.jar
Progressive Bosses/Insane96MCP/ProgressiveBosses-1.5.4-mc1.12.x.jar
Quality Tools/Tmtravlr/QualityTools-1.0.7_for_1.12.2.jar
Recurrent Complex/Ivorforce/RecurrentComplex-1.4.8.2.jar
Roguelike Dungeons/Greymerk/RoguelikeDungeons-1.12.2-1.8.0.jar
Scaling Health/SilentChaos512/ScalingHealth-1.12.2-1.3.42+147.jar
Serene Seasons/TheAdubbz/SereneSeasons-1.12.2-1.2.18-universal.jar
Shadowfacts' Forgelin/ShadowfactsDev/Forgelin-1.8.4.jar
Silent Lib/SilentChaos512SilentLib-1.12.2-3.0.14+168.jar
Sound Filters/Tmtravlr/SoundFilters-0.12.1_for_1.12.jar
Spartan Shields/ObliviousSpartan/SpartanShields-1.12.2-1.5.4.jar
Spartan Weaponry/ObliviousSpartan/SpartanWeaponry-1.12.2-beta-1.3.8.jar
Tough As Nails/TheAdubbz/ToughAsNails-1.12.2-3.1.0.139-universal.jar
Traveler's Backpack/Tiviacz1337/TravelersBackpack-1.12.2-1.0.35.jar
Traveler's Backpack IntegrationsTiviacz1337/TravelersBackpackIntegration-1.12.2-1.0.1.jar
VanillaFix/Runemoro/VanillaFix-1.0.10-150.jar
Wall-Jump! [FORGE]/genandnic/walljump-1.12.2-1.3.2.jar
Wolf Armor and Storage/satyrnidae/wolfarmor-1.12.2-3.7.5-universal-signed.jar
XP Tome/bl4ckscor3/xptome-1.12.2-v2.0.jar
